### ISSUE TITLE
IDL Metadata & Versioning

### ISSUE THREAD

**Issue by realbuffalojoe** [coral-xyz/anchor#2452]
Anchor's IDLs currently don't have support for versioning of the Anchor or IDL-gen library used to create the IDL.

As IDLs continue to evolve in the Solana ecosystem, and become a more integral piece of the program development lifecycle, it would be useful to know which version of Anchor a program was built using, and - more specifically - which version of `anchor-idl` the IDL was created with. 

This could one day help with things like CI/CD, security, breaking changes, etc. 

Overall I wouldn't consider this high priority, but it's worth considering if we're taking apart anything within the IDL scope

**Comment by callensm** [comment]
so this is referring to embedding details of the build tool versions into the IDL, not the program crate and IDL version itself (which is already done), correct?

**Comment by Aursen** [comment]
I think maybe it can be good to have a new section like shank:
`
metadata: {
     origin: "anchor",
     toolVersion: "0.27.0",
     libVersion: "0.1.0"
}
`

**Comment by realbuffalojoe** [comment]
Yeah I would say align with Shank, but we can also see where [talks on IDL Interface Standard](https://forum.solana.com/t/srfc-00008-idl-standard/66) go

**Comment by Henry-E** [comment]
Also #3644
